### ISSUE TITLE
Change Architecture to Make Supervisors Connected with Cases Rather than Volunteers

### ISSUE THREAD

**Issue by CovenantHuman** [rubyforgood/casa#3273]
## What type(s) of user does this feature affect?
 - supervisors
 - admins

## Description
Currently our architecture has supervisors having volunteers which then have cases. We would like to change things so that a supervisor has_many cases and volunteers are then discoverable through cases. This will allow us to do things like assign a single volunteer to multiple supervisors through different cases, as is currently an issue for one of our CASAs. For example, for one of our CASAs, the cases belong to the supervisors, so if a volunteer has two different cases, they may belong to two different supervisors:

```
Sup1        Sup2
  \         /
Case1    Case2
   \      /
  Volunteer
```

## QA Login Details
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

**Comment by littleforest** [comment]
Initial plan:

- [ ] Add `supervisor_id` reference to `casa_cases` table
- [ ] Write after party task that fills in the correct supervisor by using the `case_assignments` table and assigning the volunteer's supervisor.
- [ ] Need to think about what needs to change for going from one supervisor to several supervisors

**Comment by GALTdea** [comment]
I'd like to work on this one!

**Comment by mussajoop** [comment]
Hello, I would like to work on that issue :)

**Comment by mussajoop** [comment]
is it ok to start working on it ?

**Comment by littleforest** [comment]
Hi @mussajoop this is actually a quite complicated ticket because it involves a major redesign of some of our database tables and models. If you would like to work on it, it might be a good idea to find some time to pair together.

**Comment by mussajoop** [comment]
hello @littleforest, a pair prog sounds good to me. I would enjoy it. Maybe we can get in touch on Slack to discuss when we'll be both available to do so?

**Comment by littleforest** [comment]
Yes, that would be great. We can coordinate on CASA slack. This will probably be a several PR ticket!

**Comment by mussajoop** [comment]
great!

**Comment by crespire** [comment]
Has there been any movement on this at all? I'd love to help on this architectural change.

**Comment by littleforest** [comment]
@crespire I think the core team needs to treat this as an epic and make some smaller tickets from it. I'm going to move this to not-ready-to-build for now, and try to get some more manageable tickets made. Thanks for reaching out, we will keep you posted.

**Comment by compwron** [comment]
Closing for inactivity — this issue has been open for a long time with no recent activity, and is most likely no longer relevant to the current state of the project (or has been overtaken by events).

If this still matters to you or has resurfaced, please feel free to:
- **Reopen** with a quick comment on why it's still needed, or
- **Open a fresh issue** describing the current behavior and what you'd like to see — much easier than picking up multi-year-old context.

Thanks for understanding 💜
